### PR TITLE
Update doxygen Chemical Kinetics/Numerics Documentation

### LINKED ISSUE #1516: Improve main page of C++ documentation
**Problem description**

While the C++ [Modules page](https://cantera.org/documentation/docs-2.6/doxygen/html/modules.html) of the doxygen documentation provides a comprehensive overview, it would be beneficial to have a 'curated' main page pointing users to essential parts of Cantera.

At the moment, the landing page is more or less empty: see [Main page](https://cantera.org/documentation/docs-2.6/doxygen/html/index.html)

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

Restructure/consolidate/correct/update doxygen documentation for the following:

- Chemical Kinetics objects 
- Numerical utilities
- Global Data
- Exception Handling

The main objective is to generate an adequate browsable structure for the [Modules tab](https://cantera.org/documentation/docs-2.6/doxygen/html/modules.html); the underlying docstrings are largely unchanged.

In addition, the **mainfile** is switched to Markdown (supported since doxygen 1.8.3 - released in 2012).

Partially addresses #1535

Closes #1516

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

**Comment by ischoegl** [comment]
@speth ... I am stopping here. I *think* that navigating the "C++ Modules" structure has become a bit easier and I also started to fill in some gaps. I have made no attempts to update sections on integrators, phases, oneD and zeroD, or some of the ungrouped Cantera capabilities.

The PR targets the structure and should help locating information based on a handful of topics. Revising of actual docstrings goes beyond the scope of the PR.

File: include/cantera/equil/MultiPhase.h [hunk 1]
```diff
@@ -17,8 +17,9 @@ namespace Cantera
 
 class ThermoPhase;
 
-//! @defgroup equilfunctions Classes and functions used for calculating
-//!     chemical equilibrium.
+//! @defgroup equilfunctions Chemical Equilibrium Calculations
+//! Classes and functions used for calculating chemical equilibrium.
+//! @ingroup equil
```
Comment (speth) [inline include/cantera/equil/MultiPhase.h]: Would it make sense to combine the `equil` and `equilfunctions` groups? I don't see a clear distinction between them.

File: include/cantera/kinetics/ReactionRateFactory.h [hunk 2]
```diff
@@ -19,15 +19,39 @@ namespace Cantera
...
+ * @defgroup falloffGroup  Falloff Parameterizations
+ * Classes implementing fall-off in reaction rate constants due to intermolecular energy
+ * transfer and derived models.
+ * @ingroup reactionGroup
+ */
+
+/**
+ * @defgroup surfaceGroup  Coverage-Dependent Rate Parameterizations
+ * Classes implementing reaction rates that involve interfaces.
+ * @ingroup reactionGroup
```
Comment (speth) [inline include/cantera/kinetics/ReactionRateFactory.h]: I think "Interface Rate Parameterizations" is a better description for this group -- sticking rates don't have any dependence on the coverages.

File: include/cantera/base/global.h [hunk 3]
```diff
@@ -102,37 +93,56 @@ void appdelete();
...
+ * @defgroup textlogs Diagnostic Output
+ * @brief Logging and generation of diagnostic output.
  *
  * Writing diagnostic information to the screen or to a file. It is often
  * useful to be able to write diagnostic messages to the screen or to a file.
- * Cantera a set of procedures for this purpose designed to write text messages
+ * Cantera defines a set of procedures for this purpose designed to write text messages
  * to the screen to document the progress of a complex calculation, such as a
  * flame simulation.
+ * @ingroup exceptions
```
Comment (speth) [inline include/cantera/base/global.h]: The hierarchy here seems a bit off to me -- logging isn't just about errors. Would the following make more sense?
- Errors & Diagnostics
  - Exceptions
  - Warnings
  - Logging

File: include/cantera/kinetics/KineticsFactory.h [hunk 4]
```diff
@@ -48,7 +51,7 @@ Kinetics* newKineticsMgr(const string& model);
 shared_ptr<Kinetics> newKinetics(const string& model);
 
 /*!
- * Create a new kinetics manager, initialize it, and add reactions
+ * @brief Create a new kinetics manager, initialize it, and add reactions.
  *
  * @param phases     Vector of phases containing species which participate in
```
Comment (speth) [inline include/cantera/kinetics/KineticsFactory.h]: For the sake of avoiding too much variety in the different ways of using Doxygen, you might consider the "autobrief" style that we use more frequently:
```cpp
//! Create a new kinetics manager, initialize it, and add reactions.
/*!
 * @param phases     Vector of phases containing species which participate in
```

File: include/cantera/zeroD/ReactorBase.h [hunk 5]
```diff
@@ -15,9 +15,9 @@ namespace Cantera
 
 //! @defgroup ZeroD Zero-dimensional reactor networks
 //!
-//! See https://cantera.org/science/reactors.html for a description of the governing
-//! equations for specific reactor types and the methods used for solving networks of
-// interconnected reactors.
+//! See [Reactor Science Section](https://cantera.org/science/reactors/reactors.html)
+//! of the %Cantera website for a description of the governing equations for specific
```
Comment (speth) [inline include/cantera/zeroD/ReactorBase.h]:
```suggestion
//! See the [Reactor Science Section](https://cantera.org/science/reactors/reactors.html)
//! of the %Cantera website for a description of the governing equations for specific
```

**Review by speth — CHANGES_REQUESTED** [review]
Thanks for working on this, @ischoegl. I think the new landing page is a much friendlier alternative to the essentially blank page that's there now. I have only a few minor suggestions.

**Comment by ischoegl** [comment]
Thanks, @speth! I believe it's all addressed. I replaced some old group labels with a more standardized `xyzGroup` (only where you had comments).

**Review by speth — APPROVED** [review]
Thanks, @ischoegl, this looks good to me.

**Pull request merged**